packages/modules/web_themes/koala/source/src/components/BaseTable.vue

```diff
@@ -28,99 +30,196 @@
+      <template v-if="props.rowExpandable" #body="rowProps: BodySlotProps<T>">
+        <q-tr
+          :key="`main-${rowProps.key}`"
+          :props="rowProps"
+          @click="onRowClick($event, rowProps.row)"
+          class="clickable"
+        >
+          <q-td auto-width>
+            <q-btn
+              dense
+              flat
+              round
+              size="sm"
+              :icon="
+                rowProps.expand ? 'keyboard_arrow_up' : 'keyboard_arrow_down'
+              "
+              @click.stop="rowProps.expand = !rowProps.expand"
+            />
+          </q-td>
+
+          <template v-for="col in rowProps.cols" :key="col.name">
+            <!-- custom body-cell slot -->
+            <template v-if="$slots[`body-cell-${col.name}`]">
+              <slot
+                :name="`body-cell-${col.name}`"
+                v-bind="{
+                  ...rowProps,
+                  col,
+                }"
+              >
+              </slot>
+            </template>
+
+            <!-- all other column data -->
+            <q-td
+              v-else
+              :props="{
+                ...rowProps,
+                col,
+                // cast necessary as field comes from q-table and is defined: field: string | ((row: any) => any);
+                value: rowProps.row[col.field as string],
+              }"
+            >
+              <!-- cast necessary as field comes from q-table and is defined: field: string | ((row: any) => any); -->
+              {{ rowProps.row[col.field as string] }}
+            </q-td>
+          </template>
+        </q-tr>
+
+        <!-- expansion row -->
+        <q-tr
+          v-show="rowProps.expand"
+          :key="`xp-${rowProps.key}`"
+          :props="rowProps"
+          class="q-virtual-scroll--with-prev"
+        >
+          <q-td colspan="100%">
+            <slot name="row-expand" v-bind="rowProps"> </slot>
+          </q-td>
+        </q-tr>
+      </template>
+
+      <!-- forward any other slots not related to table  e.g top search field -------------------->
       <template
-        v-for="(_, name) in $slots"
-        :key="name"
-        v-slot:[name]="slotData"
+        v-for="slotName in forwardedSlotNames"
+        :key="slotName"
+        v-slot:[slotName]="slotProps"
       >
-        <slot :name="name" v-bind="slotData"></slot>
+        <slot :name="slotName" v-bind="slotProps"></slot>
       </template>
     </q-table>
   </div>
 </template>
 
-<script setup lang="ts">
-import { computed, ComputedRef } from 'vue';
-import { QTableColumn, QTableProps } from 'quasar';
+<script setup lang="ts" generic="T extends Record<string, unknown>">
+import { computed, ComputedRef, ref, useSlots } from 'vue';
+import type { QTableColumn, QTableProps } from 'quasar';
+import { BodySlotProps } from 'src/components/models/table-model';
+import { columnConfiguration } from 'src/components/models/table-model';
```

Comment:
Gibt es hier nicht bereits Definitionen in Quasar, welche wir einbinden können?
Ansonsten lässt sich der Import zusammenfassen:
```suggestion
import { BodySlotProps, columnConfiguration } from 'src/components/models/table-model';
```

Reply:
Nein, ist noch nicht angepasst.